src/kernels/piecewisepolynomial.jl

```diff
@@ -0,0 +1,43 @@
+"""
+    PiecewisePolynomialKernel(P::AbstractMatrix)
+
+Piecewise Polynomial covariance function with compact support, v = 0,1,2,3.
+The kernel functions are 2v times continuously differentiable and the corresponding
+processes are hence v times  mean-square differentiable. The kernel function is:
+```math
+    κ(x,y) = max(1-r,0)^(j+v) * f(r,j) with j = floor(D/2)+v+1
+```
+where r is the Mahalanobis distance sqrt(maha(x,z)).
+
+"""
+struct PiecewisePolynomialKernel{T1<:Integer, T2<:Real, A<:AbstractMatrix{T2}} <: BaseKernel
+    v::T1
+    maha::A
+    function PiecewisePolynomialKernel(v::T1, maha::AbstractMatrix{T2}) where {T1<:Integer,T2<:Real}
+        if v ∉ [0,1,2,3] error("Invalid paramter v=$(v). Should be 0,1,2 or 3.") end
+        LinearAlgebra.checksquare(maha)
+        new{T1,T2,typeof(maha)}(v,maha)
+    end
+end
+
+function _f(κ, r, j)
+    if κ.v==0
+        return 1
+    elseif κ.v==1
+        return 1 + (j+1)*r
+    elseif κ.v==2
+        return 1 + (j+2)*r +   (j^2+ 4*j+ 3)/ 3*r.^2
+    elseif κ.v==3
+        return 1 + (j+3)*r +   (6*j^2+36*j+45)/15*r.^2 + (j^3 + 9*j^2 + 23*j + 15)/15 * r.^3
+    else
+        error("Invalid paramter v=$(v). Should be 0,1,2 or 3.")
+    end
+end
+
+function kappa(κ::PiecewisePolynomialKernel, r::T) where {T<:Real} 
+    j = floor(size(r,2)/2 + κ.v + 1)
```

Comment:
I'd recommend using
```suggestion
    j = div(size(r, 2), 2) + κ.v + 1
```
instead.